scripts/aem.js

```diff
@@ -140,565 +302,201 @@ function setup() {
-    const blockWrapper = block.parentElement;
-    blockWrapper.classList.add(`${shortBlockName}-wrapper`);
-    const section = block.closest('.section');
-    if (section) section.classList.add(`${shortBlockName}-container`);
-  }
-}
+    this.values = options.mapValues ? new Map() : [];
 
-/**
- * Decorates all blocks in a container element.
- * @param {Element} main The container element
- */
-function decorateBlocks(main) {
-  main.querySelectorAll('div.section > div > div').forEach(decorateBlock);
-}
+    const id = this.tagName.toLowerCase();
 
-/**
- * Loads a block named 'header' into header
- * @param {Element} header header element
- * @returns {Promise}
- */
-async function loadHeader(header) {
-  const headerBlock = buildBlock('header', '');
-  header.append(headerBlock);
-  decorateBlock(headerBlock);
-  return loadBlock(headerBlock);
-}
+    const shadowRoot = this.attachShadow({ mode: 'open' });
 
-/**
- * Loads a block named 'footer' into footer
- * @param footer footer element
- * @returns {Promise}
- */
-async function loadFooter(footer) {
-  const footerBlock = buildBlock('footer', '');
-  footer.append(footerBlock);
-  decorateBlock(footerBlock);
-  return loadBlock(footerBlock);
-}
+    const template = document.getElementById(id);
 
-/**
- * Load LCP block and/or wait for LCP in default content.
- * @param {Array} lcpBlocks Array of blocks
- */
-async function waitForLCP(lcpBlocks) {
-  const block = document.querySelector('.block');
-  const hasLCPBlock = block && lcpBlocks.includes(block.dataset.blockName);
-  if (hasLCPBlock) await loadBlock(block);
-
-  document.body.style.display = null;
-  const lcpCandidate = document.querySelector('main img');
-
-  await new Promise((resolve) => {
-    if (lcpCandidate && !lcpCandidate.complete) {
-      lcpCandidate.setAttribute('loading', 'eager');
-      lcpCandidate.addEventListener('load', resolve);
-      lcpCandidate.addEventListener('error', resolve);
-    } else {
-      resolve();
+    if (template) {
+      shadowRoot.appendChild(template.content.cloneNode(true));
     }
-  });
-}
 
-init();
-
-export {
-  buildBlock,
-  createOptimizedPicture,
-  decorateBlock,
-  decorateBlocks,
-  decorateButtons,
-  decorateIcons,
-  decorateSections,
-  decorateTemplateAndTheme,
-  fetchPlaceholders,
-  getMetadata,
-  loadBlock,
-  loadBlocks,
-  loadCSS,
-  loadFooter,
-  loadHeader,
-  loadScript,
-  readBlockConfig,
-  sampleRUM,
-  setup,
-  toCamelCase,
-  toClassName,
-  updateSectionsStatus,
-  waitForLCP,
-};
+    const slots = this.querySelectorAll('[slot="item"]');
+
+    slots.forEach((element) => {
+      if (options.mapValues) {
+        const [key, value] = element.children;
+        this.values.set(key.innerText, value.innerHTML);
+      } else {
+        this.values.push(element);
+      }
+    });
+
+    this.innerHTML = '';
+
+    // Set up MutationObserver to detect changes in child nodes
+    this.observer = new MutationObserver((event) => {
```

Comment:
We can use a mutation observer in the web component to decorate inner elements like button and icons when new slots (content) are appended to the Block's shadow DOM